### PR TITLE
Print - Able to get the legend in metadata

### LINKED ISSUE #2980: 2.2 - Static legends are not taken into account when printing
I have a layer with a static legend image, thus one which is not called through a getLegendGraphic request.

When printing using this layer, this static legend is not taken into account, but the getLegendGraphic is passed onto mapfish print (in the layertree who can see the static legend, which would have http://localhost/michael/wsgi/static-cgxp/ba9db0562ce0401b8963629bbe3746e3/images/legends/legend_poi_administration.png as link):

static_legend

### PR DESCRIPTION
**Pull request by ger-benjamin**
Fix #2980 

You can temporarily try here : https://testgmf.sig.cloud.camptocamp.net/bge/s/ocFN (with the theme "demo", group "OSM functions mixed" activate layers "osm open" and, or "Cant. routes for exceptional loads"). Then print with the legend. You should get the same legend image in the print than in the layertree (both layers have have static-image).

Problem: I got a red square instead of my image. But (now) the url looks good in the print report request.

### CODE REVIEW

**Comment by ger-benjamin** [comment]
Travis error related to: https://github.com/jsdoc3/jsdoc/issues/1438

File: contribs/gmf/src/directives/print.js [hunk 1]
```diff
@@ -985,16 +995,18 @@ gmf.PrintController.prototype.getLegend_ = function(scale) {
   // For each visible layer in reverse order, get the legend url.
   layers.reverse().forEach((layer) => {
     classes = [];
-
     if (layer.getVisible() && layer.getSource()) {
       // For WMTS layers.
       if (layer instanceof ol.layer.Tile) {
-        layerName = layer.get('layerNodeName');
-        icons = this.ngeoLayerHelper_.getWMTSLegendURL(layer);
+        layerName = `${layer.get('layerNodeName')}`;
```
Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Why this change?

File: contribs/gmf/src/directives/print.js [hunk 2]
```diff
@@ -1029,6 +1044,27 @@ gmf.PrintController.prototype.getLegend_ = function(scale) {
...
 
 /**
+ * Return the metadata legendImage of a layer from the found corresponding node
+ * or undefined.
+ * @param {string} layerName a layer name.
+ * @return {string|undefined} The legendImage or undefined.
+ * @private
+ */
+gmf.PrintController.prototype.getMetadataLegendImage_ = function(layerName) {
+  const groupNode = gmf.Themes.findGroupByLayerNodeName(this.currentThemes_, `${layerName}`);
```
Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Why `${layerName}` instance of layerName

File: contribs/gmf/src/directives/print.js (hunk 1 above)
Comment (ger-benjamin) [reply contribs/gmf/src/directives/print.js]: Because closure-compiler want a string below in some function. And the "get" here can potentially give me other things.

Comment (sbrunner) [reply contribs/gmf/src/directives/print.js]: ok :-)

**Comment by ger-benjamin** [comment]
Problem: I got a red square instead of my image. But (now) the url looks good in the print report request.
